components/src/jsMain/kotlin/dev/fritz2/components/datatable/component.kt

```diff
@@ -403,7 +403,7 @@ open class DataTableComponent<T, I>(val dataStore: RootStore<List<T>>, protected
                 styling()
             }) {
                 component.stateStore.renderingRowsData(component)
-                    .renderEach(IndexedValue<T>::hashCode) { (index, rowData) ->
+                    .renderEach({ rowIdProvider(it.value) }) { (index, rowData) ->
```

Comment:
Unfortunately this does not work! We have to consider that a possible activated sorting will change the index, which must be reflected by ``renderEach``. 

I have tried to combine both approaches as follows:
```kotlin
val indexedRowIdProvider: (IndexedValue<T>) -> Int = { (index, value) -> rowIdProvider(value).hashCode() * 31 + index.hashCode() }
```
Unfortunately this also has improved the situation of the editable DataTable examples within the KitchenSink only partly!

In fact it is possible that we must delegate more responsability on top of the client for providing an efficient yet sufficient ``rowIdProvider``, which reflects the need of the table:
- often the _id_ of the model is sufficient
- if all / many columns are editable, their property counterparts need probably to be included also

We must investigate those concerns and find a solution and also probably some **recipes** for the users.